cpp/include/raft/neighbors/detail/ivf_pq_compute_similarity-inl.cuh

```diff
@@ -448,11 +456,18 @@ __global__ void compute_similarity_kernel(uint32_t n_rows,
       OutT score = kDummy;
       bool valid = i < n_samples;
       if (valid) {
```

Comment:
A neatpick: maybe put `valid && sample_filter(query_ix, label, i)` together to reduce the nesting level?

Reply:
I'll update the PR.